backend/server/issue.go

```diff
@@ -40,6 +42,16 @@ func (s *Server) registerIssueRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to create issue").SetInternal(err)
 		}
 
+		if s.MetricReporter != nil {
```

Comment:
In a separate PR, I would suggest creating a dummy metric reporter which does nothing,  so that we don't need to do null check everywhere (and avoid panic if we forget).